### PR TITLE
Fix tests on windows

### PR DESCRIPTION
**Pull request by asashour**
This PR fixes all tests on Windows.

The main reason is EOL.

### CODE REVIEW

File: community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala [hunk 1]
```diff
@@ -358,5 +360,5 @@ assert_that(
     StatementConverters.toPlannerQuery(ast.asInstanceOf[Query], table)
   }
 
-  private def parseForRewriting(queryText: String) = parser.parse(queryText.replace("\r\n", "\n"), Neo4jCypherExceptionFactory(queryText, None))
+  private def parseForRewriting(queryText: String) = parser.parse(queryText.replace(Properties.lineSeparator, "\n"), Neo4jCypherExceptionFactory(queryText, None))
```
Comment (chrisvest) [inline community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala]: All these `replace` calls transforms the strings to consistently use `\n`. They should keep replacing `\r\n` strings. Otherwise on other platforms you turn this into replacing `\n` with `\n`.

File: community/values/src/test/java/org/neo4j/values/storable/TimeZonesTest.java [hunk 2]
```diff
@@ -73,14 +73,14 @@ void weSupportDeletedZoneIdEastSaskatchewan()
...
-     * p>
+     * <p>
      * If your changes were legit, please change the expected byte[] below.
      */
     @Test
     void tzidsOrderMustNotChange() throws URISyntaxException, IOException
     {
         Path path = Paths.get( TimeZones.class.getResource( "/TZIDS" ).toURI() );
-        byte[] timeZonesInfo = Files.readAllBytes( path );
+        String timeZonesInfo = Files.readString( path ).replace( System.lineSeparator(), "\n" );
```
Comment (chrisvest) [inline community/values/src/test/java/org/neo4j/values/storable/TimeZonesTest.java]: Does Git change the line endings of `TZIDS` on checkout? Or why is this necessary? I'm suspicious about this one because reading the file as a string just like that is relying on platform encoding, and I'm not sure all encodings will let the sha256 of the file stay the same.

File: community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala (hunk 1 above)
Comment (asashour) [reply community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala]: I understand, but there should not be a case of *nix system to have `\r\n`, and replacing `\n` with `\n` would do nothing (which is not harmful).

The Git on Windows by default makes the file (so the string) with system-specific EOL.

File: community/values/src/test/java/org/neo4j/values/storable/TimeZonesTest.java (hunk 2 above)
Comment (asashour) [reply community/values/src/test/java/org/neo4j/values/storable/TimeZonesTest.java]: Yes, the EOL of `TZIDS` are system-specific, at least on Windows. We can change that with `.gitattributes` if it is better.

The current `sha256` seems to be successful on *nix system, and with this PR the test passes with the exact `sha256` on windows, so I believe it is the same on all systems, since it only depends on the text + `\n`, and not on EOL.

**Comment by klaren** [comment]
Are the tests failing? We run the tests on windows all the time without an issue.

**Comment by asashour** [comment]
Really? the tests are green on Windows?

Yes, the fixes were done, to make maven passes all the tests.

**Comment by asashour** [comment]
Possibly it will pass, when checking the source on Windows, the files are configured to get `\n` on the build machine, but this is not the default `git` checkout (without any local configuration).

**Comment by klaren** [comment]
That might the case, I'm going to investigate that further. 

There is a green build from a couple of minutes ago, windows server 2019, oracle jdk 11, mave 3.6.1.

**Comment by klaren** [comment]
You're right, we use checkout-as-is so we do not observe the errors here. Thanks for the fix! I'm pushing it through our pipeline now.

File: community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala (hunk 1 above)
Comment (klaren) [reply community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala]: I would argue that replacing `\r\n\` with `\n` on *nix system is not harmful either. So I think we should leave it as is.

Comment (asashour) [reply community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala]: Well, it should be changed:

- Mainly to make the tests pass on other systems which uses `\r` only (Mac OS)
- To be uniform with newly added changes, otherwise we should also change all places to `replace("\r\n", "\n")`, ignoring Mac OS.

Comment (klaren) [reply community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala]: Mac OS has not used `\r` since 2001 so I think we are safe :wink:

File: annotations/src/main/java/org/neo4j/annotations/api/PublicApiAnnotationProcessor.java [hunk 3]
```diff
@@ -148,7 +148,7 @@ private void generateSignature() throws IOException
             StringBuilder sb = new StringBuilder();
             for ( final String element : publicElements )
             {
-                sb.append( element ).append( '\n' );
+                sb.append( element ).append( System.lineSeparator() );
```
Comment (klaren) [inline annotations/src/main/java/org/neo4j/annotations/api/PublicApiAnnotationProcessor.java]: Our build system did not like this change, we compare to the files in `annotations/src/test/resources/signatures` which will have `\n` in our case. I'm not sure how to address that.

**Review by klaren — CHANGES_REQUESTED** [review]

File: community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala (hunk 1 above)
Comment (asashour) [reply community/cypher/cypher-planner/src/test/scala/org/neo4j/cypher/internal/compiler/planner/logical/OptionalMatchRemoverTest.scala]: Thanks for info.

Change reverted.

All other places were changed to `.replace("\r\n", "\n")`

File: annotations/src/main/java/org/neo4j/annotations/api/PublicApiAnnotationProcessor.java (hunk 3 above)
Comment (asashour) [reply annotations/src/main/java/org/neo4j/annotations/api/PublicApiAnnotationProcessor.java]: Well, in this case I reverted this change, and added `.gitattributes` to ensure all `PublicApi.txt` are ending with `\n`, hope it is fine now. However, I would suggest to change the `checkout-as-is`, if possible.

I will fully test it later locally, by recloning the repository, since local `.gitattributes` changes are not immediately followed.

Comment (klaren) [reply annotations/src/main/java/org/neo4j/annotations/api/PublicApiAnnotationProcessor.java]: You only need the git attributes for the files in the `annotations/src/test/resources/signatures` and `annotations/src/test/resources/META-INF/services` directories.

**Comment by asashour** [comment]
~~The current changes are just the ones from `4.0` branch (after being verified to fail a test case).~~

~~However, there are other things, which are to be handled separately.~~

1 - ~~`community\graphdb-api\PublicApi.txt`  actually differs in content to `community\graphdb-api\target\classes\META-INF\PublicApi.txt`, the first one has few more lines:~~


2- ~~With `maven`, the build hangs in `neo4j-kernel-api` module, and it seems `PrivilegeActionTest` and `IndexConfigProviderTest` run in parallel, and block each other.~~

3- ~~CvsImporterTest fail~~

**Comment by chrisvest** [comment]
Make sure you are on the very latest Java 11, so you're not affect by https://bugs.openjdk.java.net/browse/JDK-8234466

**Comment by asashour** [comment]
Thanks for the hint, upgrading to Java `11.0.7` makes it running.

**Comment by asashour** [comment]
Finally, the build is successful, thanks.
